examples/fakegroundstation/ground_station_service.py

```diff
@@ -0,0 +1,136 @@
+import time
+from concurrent import futures
+
+import grpc
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from stellarstation.api.v1.groundstation import groundstation_pb2_grpc
+from stellarstation.api.v1.groundstation import groundstation_pb2
+
+
+SECONDS_BEFORE_PLAN_START = 10
+PLAN_DURATION_SECONDS = 600
+CURRENT_PLAN_ID = "3"
+
+
+class GroundStationServiceServicer(groundstation_pb2_grpc.GroundStationServiceServicer):
+    def ListPlans(self, request, context) -> groundstation_pb2.ListPlansResponse:
+        """Lists the plans for a particular ground station.
+
+        The request will be closed with an `INVALID_ARGUMENT` status if `ground_station_id`,
+        `aos_after`, or `aos_before` are missing, or the duration between the two times is longer than
+        31 days.
+        """
+        print('Got request for ListPlans')
+        if request.ground_station_id is None or request.ground_station_id == "":
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details('Ground station ID not set')
+            raise RuntimeError('Ground station ID not set')
+        if request.aos_after is None:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details('AOS after not set')
+            raise RuntimeError('AOS after not set')
+        if request.aos_before is None:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details('AOS before not set')
+            raise RuntimeError('AOS before not set')
+
+        aos_after = request.aos_after.seconds + request.aos_after.nanos / 10.**9
+        aos_before = request.aos_before.seconds + request.aos_before.nanos / 10. ** 9
+
+        if aos_before - aos_after > 31 * 24 * 3600:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details('Duration between aos_after and aos_before > 31 days')
+            raise RuntimeError('Duration between aos_after and aos_before > 31 days')
+
+        now = time.time()
+
+        satellite_coordinates = [
+            groundstation_pb2.SatelliteCoordinates(
+                time=Timestamp(seconds=int(now + SECONDS_BEFORE_PLAN_START + i)),
+                range_rate=(2.1e7 + i * 1e4)
+            )
+            for i in range(PLAN_DURATION_SECONDS)
+        ]
+        # TODO: Fill in the other fields of plan
+        response = groundstation_pb2.ListPlansResponse(
+            plan=[groundstation_pb2.Plan(
+                plan_id=CURRENT_PLAN_ID,
+                satellite_coordinates=satellite_coordinates)]
+        )
+        return response
+
+    def OpenGroundStationStream(self, request_iterator, context):
```

Comment:
Snake case. Since this function is in GroundStationServiceServicer, open_stream might be enough as a name withoug "GroundStation".

Reply:
These are overriding the methods of the `groundstation_pb2_grpc.GroundStationServiceServicer` stub.